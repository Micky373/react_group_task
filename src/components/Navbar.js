import React from 'react';
import { Link } from 'react-router-dom';
import Img from '../images/space_travelers.png';

const Navbar = () => (
  <nav>
    <div className="nav_container">
      <div className="image_container">
        <img alt="header logo" src={`${Img}`} />
        <h1>Check Travlers&apos; Hub</h1>
      </div>
      <ul className="nav_links">
        <Link to="/"><li className="page_links">Rockets</li></Link>
        <Link to="/Missions"><li className="page_links">Missions</li></Link>
        <div className="line_end" />
        <Link to="/MyProfile"><li className="page_links">My profile</li></Link>
      </ul>
    </div>
    <div className="nav_end" />
  </nav>
);

export default Navbar;
