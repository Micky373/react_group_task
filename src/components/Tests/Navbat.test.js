import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Navbar from '../Navbar';
import store from '../../redux/configureStore';

it('it works', () => {
  const tree = render(
    <Provider store={store}>
      <Router>
        <Navbar />
        <Routes>
          <Route exact path="/" />
          <Route path="/MyProfile" />
          <Route path="/Missions" />
        </Routes>
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  render(
    <Provider store={store}>
      <Router>
        <Navbar />
        <Routes>
          <Route exact path="/" />
          <Route path="/MyProfile" />
          <Route path="/Missions" />
        </Routes>
      </Router>
    </Provider>,
  );
  expect(screen.getByText(/Space/)).toBeInTheDocument();
});
