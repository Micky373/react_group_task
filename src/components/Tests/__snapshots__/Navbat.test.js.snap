// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <div
          class="nav_container"
        >
          <div
            class="image_container"
          >
            <img
              alt="header logo"
              src="space_travelers.png"
            />
            <h1>
              Space Travlers' Hub
            </h1>
          </div>
          <ul
            class="nav_links"
          >
            <a
              href="/"
            >
              <li
                class="page_links"
              >
                Rockets
              </li>
            </a>
            <a
              href="/Missions"
            >
              <li
                class="page_links"
              >
                Missions
              </li>
            </a>
            <div
              class="line_end"
            />
            <a
              href="/MyProfile"
            >
              <li
                class="page_links"
              >
                My profile
              </li>
            </a>
          </ul>
        </div>
        <div
          class="nav_end"
        />
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <div
        class="nav_container"
      >
        <div
          class="image_container"
        >
          <img
            alt="header logo"
            src="space_travelers.png"
          />
          <h1>
            Space Travlers' Hub
          </h1>
        </div>
        <ul
          class="nav_links"
        >
          <a
            href="/"
          >
            <li
              class="page_links"
            >
              Rockets
            </li>
          </a>
          <a
            href="/Missions"
          >
            <li
              class="page_links"
            >
              Missions
            </li>
          </a>
          <div
            class="line_end"
          />
          <a
            href="/MyProfile"
          >
            <li
              class="page_links"
            >
              My profile
            </li>
          </a>
        </ul>
      </div>
      <div
        class="nav_end"
      />
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
